moment_kinetics/src/gauss_legendre.jl

```diff
@@ -906,6 +913,26 @@ function setup_global_weak_form_matrix!(QQ_global::Array{mk_float,2},
             QQ_global[end,:] .= 0.0
             QQ_global[end,end] = 1.0
         end
+        # requires RHS vector b[1],b[end] = boundary values
+    end
+    if periodic_bc
+        # Make periodic boundary condition by modifying elements of matrix for duplicate point
+        # add assembly contribution to lower endpoint from upper endpoint
+        j = coord.nelement_local
+        get_QQ_local!(QQ_j,j,lobatto,radau,coord,option)
+        iminl = imin[j] - (coord.nelement_local > 1)
+        imaxl = imax[j]
+        QQ_global[1,iminl:imaxl] .+= QQ_j[end,:]
+        # Enforce continuity at the periodic boundary
+        # All-zero row in RHS matrix sets last element of `b` vector in
+        # `mass_matrix.x = b` to zero
+        QQ_global[end,:] .= 0.0
+        if option == "M"

```

Comment:
This if statement should not be here: if it is then we cannot use the periodic boundary conditions for solving ODEs.

Reply:
I think I understand that the intention here is for this whole block to handle periodic BC for both the K and M matrices. However, there are two separate ways to use these matrices. If we want to solve an ODE, then the role of K and M is reversed, meaning that we need an M that is just multiplied (not LU decomposed), and a K that has BCs. This needs modification before it can go forward -- it is not general enough to be used.

Reply:
I was thinking about that a bit - I think we might be able to treat K and M the same. Since we know that in the periodic case that `x[1] == x[end]`, then if the last row of the matrix is (1 0 0 ... 0 0 -1) then the last entry of `K.x` or `M.x` would still be zero, without having to zero out the whole of the last row. I haven't tested yet, but if that works it would mean we could use M and K either way around without any more special handling.

Reply:
I've checked that this suggestion does work, so it's not necessary to distinguish between 'lhs' and 'rhs' matrices - I've just pushed a commit returning the 2 flags to one `periodic_bc`.

Reply:
Nice! I think in practice one might still need to set b[end] = 0 exactly (b from K.x = b), but I am happy to leave the code as it is for now. Thanks for taking look.